pymatgen/io/lobster/inputs.py

```diff
@@ -116,74 +116,80 @@ class Lobsterin(UserDict, MSONable):
+
+    def __init__(self, settingsdict: dict) -> None:
         """
         Args:
             settingsdict: dict to initialize Lobsterin.
         """
         super().__init__()
-        # check for duplicates
-        keys = [key.lower() for key in settingsdict]
+
+        # Check for duplicates from case sensitivity
+        keys = tuple(map(str.lower, settingsdict.keys()))
         if len(keys) != len(set(keys)):
             raise KeyError("There are duplicates for the keywords!")
+
         self.update(settingsdict)
 
-    def __setitem__(self, key, val) -> None:
+    def __setitem__(self, key, val: Any) -> None:
         """
-        Add parameter-val pair to Lobsterin. Warns if parameter is not in list of
-        valid lobsterin tags. Also cleans the parameter and val by stripping
-        leading and trailing white spaces. Similar to INCAR class.
+        Add key-value pairs. Necessary due to the missing
+        case sensitivity of LOBSTER. Also clean the keys
+        and values by stripping leading and trailing white spaces.
+
+        Raises:
+            KeyError: if parameter is not available.
         """
-        # due to the missing case sensitivity of lobster, the following code is necessary
-        new_key = next((key_here for key_here in self if key.strip().lower() == key_here.lower()), key)
+        new_key = key.strip().lower()
 
-        if new_key.lower() not in [element.lower() for element in Lobsterin.AVAILABLE_KEYWORDS]:
-            raise KeyError("Key is currently not available")
+        if new_key not in self.AVAILABLE_KEYWORDS:
+            raise KeyError(f"Key {new_key} is currently not available")
 
         super().__setitem__(new_key, val.strip() if isinstance(val, str) else val)
 
     def __getitem__(self, key) -> Any:
-        """Implements getitem from dict to avoid problems with cases."""
-        normalized_key = next((k for k in self if key.strip().lower() == k.lower()), key)
-
-        key_is_unknown = normalized_key.lower() not in map(str.lower, Lobsterin.AVAILABLE_KEYWORDS)
-        if key_is_unknown or normalized_key not in self.data:
-            raise KeyError(f"{key=} is not available")
+        """To avoid cases sensitivity problems."""
+        try:
+            return self.data[key.strip().lower()]
 
-        return self.data[normalized_key]
+        except KeyError as exc:
+            raise KeyError(f"{key=} is not available") from exc
 
     def __contains__(self, key) -> bool:
-        """Implements getitem from dict to avoid problems with different key casing."""
-        normalized_key = next((k for k in self if key.strip().lower() == k.lower()), key)
+        """To avoid cases sensitivity problems."""
+        return key.lower().strip() in self.data
 
-        key_is_unknown = normalized_key.lower() not in map(str.lower, Lobsterin.AVAILABLE_KEYWORDS)
-        return not key_is_unknown and normalized_key in self.data
```

Comment:
This doesn't look necessary, user cannot set a key that is not in `AVAILABLE_KEYWORDS` because it has already been guarded by `__setitem__`